code/game/gamemodes/traitor/traitor.dm

```diff
@@ -138,11 +138,6 @@
 					maroon_objective.owner = traitor
 					maroon_objective.find_target()
 					traitor.objectives += maroon_objective
-				else

```

Comment:
This won't even work well code wise.

Reply:
Ah, yes - if you don't guarantee an objective is added inside the loop, strange things will happen in terms of objective assignment - resulting in less objectives being assigned overall.